### PR TITLE
Laboratory 2

### PR DESCRIPTION
**Pull request by olehio-p**

### CODE REVIEW

File: src/projects/observers.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class ProjectObserver(Observer):
+    def __init__(self, recipient: str):
+        self._recipient = recipient
+
+    @property
+    def recipient(self):
+        return self._recipient
+
+    def update(self, message: str):
```
Comment (dmytro-maherovskyi-lpnu) [inline src/projects/observers.py:18]: I believe this method name should be 'notify'

File: src/projects/project.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
+from datetime import datetime
+from typing import Optional
+from .observers import Observable
+
+
+class TaggingMixin:
```
Comment (dmytro-maherovskyi-lpnu) [inline src/projects/project.py:6]: What is Mixin?

File: src/projects/project.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+    def remove_tag(self, tag: str):
+        if tag in self._tags:
+            self._tags.remove(tag)
+
+
+class TimestampMixin:
+    def __init__(self):
+        super().__init__()
+        self._created_at = datetime.now()
```
Comment (dmytro-maherovskyi-lpnu) [inline src/projects/project.py:30]: Usually we are tend to use utc now for time stamps - because this code behaves differently in different timezones

**Review by dmytro-maherovskyi-lpnu — APPROVED** [review]

File: src/projects/observers.py (hunk 1 above)
Comment (olehio-p) [reply src/projects/observers.py:18]: I don't quiet understand why. I created this class based on observer pattern by GoF and according to them this method should be named as 'update'

File: src/projects/project.py (hunk 2 above)
Comment (olehio-p) [reply src/projects/project.py:6]: A bit hard to explain using comments:(

File: src/projects/project.py (hunk 3 above)
Comment (olehio-p) [reply src/projects/project.py:30]: Got it!

File: src/projects/observers.py (hunk 1 above)
Comment (dmytro-maherovskyi-lpnu) [reply src/projects/observers.py:18]: resolved

File: src/projects/project.py (hunk 2 above)
Comment (dmytro-maherovskyi-lpnu) [reply src/projects/project.py:6]: resolved

**Review by dmytro-maherovskyi-lpnu — APPROVED** [review]

**Review by dmytro-maherovskyi-lpnu — APPROVED** [review]

**Pull request merged**